cmd/clusterctl/clusterdeployer/clusterclient/clusterclient.go

```diff
@@ -354,6 +354,7 @@ func (c *client) GetMachineDeploymentsForCluster(cluster *clusterv1.Cluster) ([]
 
 	var machineDeployments []*clusterv1.MachineDeployment
 	for _, md := range machineDeploymentList.Items {
+		md := md
```

Comment:
Is this actually effective? We should probably use `md := machineDeploymentList.Items[i]` instead?

Reply:
yeah, `md := machineDeploymentList.Items[i]` looks  more clearer